extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/dashmanifestcreators/YoutubeDashManifestCreatorsUtils.java

```diff
@@ -0,0 +1,856 @@
+        final DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        try {
+            documentBuilderFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+            documentBuilderFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+        } catch (final Exception ignored) {
+            // Ignore exceptions as setting these attributes to secure XML generation is supported
+            // by all platforms (like the Android implementation)
+        }
+
+        final DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+        return documentBuilder.newDocument();
+    }
+
+    /**
+     * Generate a new {@link TransformerFactory} secured from XXE attacks, on platforms which
+     * support setting {@link XMLConstants#ACCESS_EXTERNAL_DTD} and
+     * {@link XMLConstants#ACCESS_EXTERNAL_SCHEMA} in {@link TransformerFactory} instances.
+     *
+     * @param document the document to convert, which must have been created using
+     *                 {@link #newDocument()} to properly prevent XXE attacks
+     * @return the document converted to an XML string, making sure there can't be XXE attacks
+     */
+    // Sonar warning is suppressed because it is still shown even if we apply its solution
+    @SuppressWarnings("squid:S2755")
+    private static String documentToXml(@Nonnull final Document document)
+            throws TransformerException {
+
+        final TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        try {
+            transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+            transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+        } catch (final Exception ignored) {
+            // Ignore exceptions as setting these attributes to secure XML generation is supported
+            // by all platforms (like the Android implementation)
+        }
+
+        final Transformer transformer = transformerFactory.newTransformer();
+        transformer.setOutputProperty(OutputKeys.VERSION, "1.0");
+        transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
+        transformer.setOutputProperty(OutputKeys.STANDALONE, "no");
+
+        final StringWriter result = new StringWriter();
+        transformer.transform(new DOMSource(document), new StreamResult(result));
+
+        return result.toString();
+    }
+
+    /**
+     * Append {@link #SQ_0} for post-live-DVR and OTF streams and {@link #RN_0} to all streams.
+     *
+     * @param baseStreamingUrl the base streaming URL to which the parameter(s) are being appended
+     * @param deliveryType     the {@link DeliveryType} of the stream
+     * @return the base streaming URL to which the param(s) are appended, depending on the
+     * {@link DeliveryType} of the stream
+     */
+    @SuppressWarnings({"checkstyle:FinalParameters", "checkstyle:FinalLocalVariable"})
```

Comment:
I think it is better not to suppress warnings and do something like this:
```java
return baseStreamingUrl + (deliveryType == DeliveryType.PROGRESSIVE ? "" : SQ_0) + RN_0;
```
or like this:
```java
String result = baseStreamingUrl;
if (deliveryType != DeliveryType.PROGRESSIVE) {
    result += SQ_0;
}
result += RN_0;
return result;
```